File: typescript/tsfind/src/argtoken.ts
/*
 * argtoken.js
 *
 * encapsulates an argument token
 */

'use strict';

import {ArgTokenType} from "./argtokentype";

export class ArgToken {
    name: string;
    type: ArgTokenType;
    value: boolean | number | string | string[];

    constructor(name: string, type: ArgTokenType, value: boolean | number | string | string[]) {
        this.name = name;
        this.type = type;
        this.value = value;
    }
}

File: java/javafind/lib/src/main/java/javafind/ArgToken.java
/*******************************************************************************
ArgToken

Class to encapsulate a command line argument token

@author Cary Clark &lt;clarkcb@gmail.com&gt;
@version $Rev$
@copyright Cary Clark 2025
*******************************************************************************/

package javafind;

public record ArgToken(String name, ArgTokenType type, Object value) {
}
